### PR TITLE
Adds a lock field to the profile throttle db.

### PR DESCRIPTION
**Pull request by aman-atakama**

### CODE REVIEW

File: policy_basics/per_profile_throttle.py [hunk 1]
```diff
@@ -164,8 +203,16 @@ def approve_request(self, request: ApprovalRequest):
         return self._approve_profile_request(request.profile.profile_id)
 
     def _approve_profile_request(self, profile_id):
-        pc = self.db.get(self.rule_id, profile_id)
+        pc = self.db.get(self.rule_id, profile_id, lock=True)
+        if pc is None:
+            log.debug(
```
Comment (higleyc) [inline policy_basics/per_profile_throttle.py:208]: log.warning

File: policy_basics/per_profile_throttle.py [hunk 2]
```diff
@@ -35,7 +37,15 @@ class ProfileCount:
     hour_cnt: int = 0
     day_cnt: int = 0
 
-    def __init__(self, timestamp=None, hour_cnt=0, day_cnt=0):
+    def __init__(
+        self,
+        timestamp=None,
+        hour_cnt=0,
+        day_cnt=0,
+        lock_value: str = None,
```
Comment (howufiger7) [inline policy_basics/per_profile_throttle.py:45]: minor: `Optional[str]`

File: tests/test_per_profile_throttle.py [hunk 3]
```diff
@@ -250,3 +283,53 @@ def test_end_to_end():
...
+    )
+    pr2 = ProfileThrottleRule(
+        {"per_day": 10, "persistent": True, "rule_id": "rid", "expiry_secs": 3}
+    )
+    assert pr1.db.lock_value != pr2.db.lock_value  # Sanity check
+
+    assert pr1._approve_profile_request(b"pid")
+    assert not pr2._approve_profile_request(b"pid"), "Expected DB to be locked"
+
+    time.sleep(3)
```
Comment (howufiger7) [inline tests/test_per_profile_throttle.py:333]: minor: would be good to make 3 a constant in this func

**Review by howufiger7 — APPROVED** [review]
a couple of nits, LGTM otherwise

**Pull request merged**